transit_odp/timetables/dataframes.py

```diff
@@ -179,6 +196,146 @@ def journey_pattern_sections_to_dataframe(sections):
+                }
+            )
+
+    return pd.DataFrame(all_vechicle_journeys)
+
+
+def operating_profile_to_df(operating_profiles, op_type):
+    serviced_orgs_refs_df = []
+    if operating_profiles is not None and op_type == "VehicleJourney":
+        for operating_profile in operating_profiles:
+            serviced_organisation_day_type = operating_profile.get_element_or_none(
+                ["ServicedOrganisationDayType"]
+            )
+            if serviced_organisation_day_type:
+                days_of_operation = serviced_organisation_day_type.get_element_or_none(
+                    ["DaysOfOperation"]
+                )
+                days_of_non_operation = (
+                    serviced_organisation_day_type.get_element_or_none(
+                        ["DaysOfNonOperation"]
+                    )
+                )
+                if days_of_operation:
+                    operational = True
+                    working_days = days_of_operation.get_element("WorkingDays")
+                elif days_of_non_operation:
+                    operational = False
+                    working_days = days_of_non_operation.get_element("WorkingDays")
+                serviced_org_ref = working_days.get_element(
+                    "ServicedOrganisationRef"
+                ).text
+
+                serviced_orgs_refs_df.append(
+                    {"serviced_org_ref": serviced_org_ref, "operational": operational}
+                )
+
+    if operating_profiles is not None and op_type == "Services":
+        for operating_profile in operating_profiles:
+            serviced_organisation_day_type = operating_profile.get_element_or_none(
+                ["ServicedOrganisationDayType"]
+            )
+            if serviced_organisation_day_type:
+                days_of_operation = serviced_organisation_day_type.get_element_or_none(
+                    ["DaysOfOperation"]
+                )
+                days_of_non_operation = (
+                    serviced_organisation_day_type.get_element_or_none(
+                        ["DaysOfNonOperation"]
+                    )
+                )
+                if days_of_operation:
+                    operational = True
+                    working_days = days_of_operation.get_element("WorkingDays")
+                elif days_of_non_operation:
+                    operational = False
+                    working_days = days_of_non_operation.get_element("WorkingDays")
+                serviced_org_ref = working_days.get_element(
+                    "ServicedOrganisationRef"
+                ).text
+
+                serviced_orgs_refs_df.append(
+                    {"serviced_org_ref": serviced_org_ref, "operational": operational}
+                )
+
+    return pd.DataFrame(serviced_orgs_refs_df)
+
+
+def serviced_organisations_to_dataframe(serviced_organisations):
+    serviced_organisations_df = []
+    if serviced_organisations_to_dataframe is not None:
```

Comment:
should this be `serviced_organisations` instead of `serviced_organisations_to_dataframe`. If `serviced_organisations` then the if condition in the extract function can be removed as one of the if conditions are redundant